### ISSUE TITLE
news state "structured covariance models now work in zero-inflation and dispersion components (GH #579) "

### ISSUE THREAD

**Issue by amynang** [glmmTMB/glmmTMB#611]
copying example from relevant vignette:
```
require(MASS)
n <- 6                                              ## Number of time points
x <- mvrnorm(mu = rep(0,n),
             Sigma = .7 ^ as.matrix(dist(1:n)) )    ## Simulate the process using the MASS package
y <- x + rnorm(n)                                   ## Add measurement noise
times <- factor(1:n, levels=1:n)
levels(times)
group <- factor(rep(1,n))
dat0 <- data.frame(y,times,group)
glmmTMB(y ~ ar1(times + 0 | group),
                  dispformula = ~ ar1(times + 0 | group), 
                  data=dat0)
```
returns:

> Error in getXReTrms(dispformula, mf, fr, ranOK = FALSE, type = "dispersion",  : no random effects allowed in dispersion term

**Comment by bbolker** [comment]
OK, the NEWS is wrong; I fixed a bug that stopped structured covariance models from working in zero-inflation components, which *would* also have stopped them from working in dispersion components.

However ... we (glmmTMB developers) have from the beginning not implemented random effects in dispersion models. This was a (somewhat) conscious decision, as we felt that it would often be unwise (or beyond the scope of most ecological data sets) to try to fit a hierarchical model of variance. For what it's worth I don't see anything intrinsically wrong with setting up a model of the form

```
y[i,j] ~ Normal(mu[i,j], exp(logsd[i]))
logsd[i] ~ Normal(logsd_0, S)
```

but I can imagine it would open a lot of statistical cans of worms. [Westneat et al.](https://link.springer.com/content/pdf/10.1007/s00265-017-2315-3.pdf) have fitted such "double hierarchical GLMs" using JAGS, and there are a reasonable number of such approaches using h-GLMs *sensu* Nelder, Lee, Pawitan ([Google scholar search](https://scholar.google.ca/scholar?hl=en&as_sdt=0%2C5&q=double+hierarchical+GLM&btnG=)).  (There is an `hglm` package in R that implements this class of models, but it only allows fixed effects on the dispersion ... don't know why that particular limitation is imposed.)

We could fairly easily relax this restriction (by removing the code that throws the error), but there are a good number of places in the code that assume that only the conditional and zero-inflation components will have random effects components - someone would have to go through and find/fix these. It would also be a *very* good idea to run a series of tests on simulated data to make sure that such models actually performed as expected (e.g., that the Laplace approximation is reasonably accurate in this context).

**Comment by amynang** [comment]
I'm sorry, I was not suggesting that there should be random effects in the dispersion component, (nor am I trying to fit such a model). I am not in a position to judge whether something like that would make sense or not. I just updated the package and going through the NEWS found the point related to #579 and it is confusing and in contradiction with the description of the package in Brooks et al. 2017 for example, so I thought to bring it up. That is the only reason I opened this issue.
